widgets/src/widgets/topStats.ts

```diff
@@ -0,0 +1,53 @@
+import { Widget } from '@/widgets/base'
+import { WIDGET_COLORS } from '@/constants'
+
+import type { WidgetConfig } from '@/widgets/base'
+import type { TopStatsWidgetParams } from '@/interfaces'
+
+export class TopStatsWidget extends Widget implements TopStatsWidgetParams {
+  color: (typeof WIDGET_COLORS.topStats)[number]
+
+  constructor(params: WidgetConfig & TopStatsWidgetParams) {
+    super(params)
+    this.color = params.color
+  }
+
+  get topStatsWidgetRequestBody(): TopStatsWidgetParams {
+    return {
+      color: this.color,
+    }
+  }
+
+  updateDefaults({ color }: Partial<TopStatsWidgetParams>) {
+    this.color = color ?? this.color
+  }
+
+  validateOptions() {
+    if (!WIDGET_COLORS.topStats.includes(this.color)) {
+      throw new Error(
+        `Greenspark - "${
+          this.color
+        }" was selected as the color for the Top Stats Widget, but this color is not available. Please use one of the available colors: ${WIDGET_COLORS.topStats.join(
+          ', ',
+        )}`,
+      )
+    }
+  }
+
+  async render(options?: Partial<TopStatsWidgetParams>, containerSelector?: string): Promise<void> {
```

Comment:
this could come maybe from the parent class with a generic? it's implemented in couple places and only the option's type changing.